src/Database/DBA.php

```diff
@@ -645,6 +645,34 @@ public static function collapseCondition(array $condition)
 		return $condition;
 	}
 
+	/**
+	 * Merges the provided conditions into a single collapsed one
+	 *
+	 * @param array ...$conditions One or more condition arrays
+	 * @return array A collapsed condition
+	 * @see DBA::collapseCondition() for the condition array formats
+	 */
+	public static function mergeConditions(array ...$conditions)
```

Comment:
Have you checked this with the two possible ways of providing conditions?

Reply:
Yes, you can see in this function that the conditions are collapsed which normalize them from the two existing forms.

Reply:
I only had a short glimpse right now.